src/main/java/com/hedvig/productPricing/campaign/service/DomainService.kt

```
@@ -238,11 +253,22 @@ class DomainService(
       campaignOwnerRepository.findByMemberId(memberId)!!
     }
 
-    campaignOwner.setInsuranceActivationDate(insuranceActivatedAt);
+    campaignOwner.setInsuranceActivationDate(insuranceActivatedAt)
 
     campaignOwnerRepository.save(campaignOwner)
   }
 
+  fun assignRedeemedCampaign(memberId: String, campaignId: UUID, incentive: Incentive) {
+    val member = campaignOwnerRepository.findByMemberId(memberId)
+    require(member != null)
+
+    val campaign = campaignRepository.findByIdOrNull(campaignId)
+    require(campaign != null)
```

Shall we throw a custom exception here?